x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialAggregateFunction.java

```diff
@@ -22,26 +25,34 @@
  * select the best one.
  */
 public abstract class SpatialAggregateFunction extends AggregateFunction {
-    protected final boolean useDocValues;
+    protected final FieldExtractPreference preference;
```

Comment:
Can you find a different name for `preference`? Something that's more explanatory.

Reply:
As in, `fieldExtractPreference`?